src/main/java/org/springframework/guice/annotation/ModuleRegistryConfiguration.java

```diff
@@ -252,69 +288,74 @@ else if (e instanceof PrivateElements) {
 	 * @return de-duplicated list of bindings
 	 */
 	protected List<Element> removeDuplicates(List<Element> elements) {
-		List<Element> duplicateElements = elements.stream().filter((e) -> e instanceof Binding)
-				.map((e) -> (Binding<?>) e)
-				.collect(Collectors.groupingBy(ModuleRegistryConfiguration::getLinkedKeyIfRequired)).entrySet().stream()
```

Comment:
The change to `getLinkedKeyIfRequired` appears to be what caused this to break when coming from 1.x, but when I pulled that thread there turned out to all these other cases too.